src/Parser/Extra.elm

```diff
@@ -0,0 +1,112 @@
+module Parser.Extra
+    exposing
+        ( between
+        , braces
+        , brackets
+        , by
+        , many
+        , parens
+        , some
+        , spacy
+        )
+
+{-| Some convenience parser combinators.
+
+
+# Combinators
+
+@docs many, some, by, spacy, between, parens, braces, brackets
+
+-}
+
+import Parser exposing (..)
+
+
+{-| Apply a parser one or more times and return a list of the results.
+-}
+some : Parser a -> Parser (List a)
+some item =
+    succeed (::)
+        |= item
+        |= many item
+
+
+{-| Apply a parser zero or more times and return a list of the results.
+-}
+many : Parser a -> Parser (List a)
+many item =
+    let
+        helper vs =
+            oneOf
+                [ succeed (\v -> Loop (v :: vs))
+                    |= item
+                , succeed ()
+                    |> map (\() -> Done (List.reverse vs))
+                ]
+    in
+    loop [] helper
+
+
+{-| Parse an item one or more times separated by a separator.
+-}
+by : Parser () -> Parser a -> Parser (List a)
+by sep item =
+    let
+        more =
+            succeed identity
+                |. backtrackable sep
+                |= item
+    in
+    succeed (::)
+        |= item
+        |= many more
+
+
+{-| Parse an item with spaces before and after.
+-}
+spacy : Parser () -> Parser ()
```

Comment:
The type signature is wrong. You probably care about the `item` between `spaces` so it should be `Parser a -> Parser a`.

Also, I'd rather have `spacy` implemented in terms of `between` for consistency:
```elm
spacy : Parser a -> Parser a
spacy =
    between spaces spaces
```

Reply:
Yeah, you're right. I'll fix this.